src/devices/src/virtio/block/device.rs

```diff
@@ -258,15 +258,34 @@ impl Block {
                             break;
                         }
                     }
+
                     let status = match request.execute(&mut self.disk, mem) {
                         Ok(l) => {
-                            len = l;
+                            // Account for the status descriptor as well.
```

Comment:
> rephrased comment
> 
> didn't use `checked_add` since `len` or `l` cannot come close to `u32::MAX` bytes. That would mean single IO `read`/`write` of at least 4GB which is unlikely/not possible in normal usecases.
> 
> In case of naughty guest driver creating requests > 4GB what is the effect of overflow here? reported `len` smaller than actually written to virtio buffers - I understand that this is allowed by the virtio protocol, right?
> 
It is allowed indeed, but I still see this as an error case, since the driver should never send a request that will cause the `len` field from the used ring overflow an `u32` (it is no more naughty, but rather faulty). Also, since we are having a very simple implementation of the block device and we always assume that there is only one data descriptor in the request (which the spec doesn't enforce in any way, but it is rather a Linux driver particularity), this `len` from here can only be <= u32::MAX. And keeping in mind the [following](https://github.com/oasis-tcs/virtio-spec/blob/master/content.tex#L4651), it seems pretty bad to accept as valid u32::MAX.

> If we were to add overflow checks just to be extra safe, I'm not sure what we could do in case of an overflow. Would we `unwrap`/`panic`? Would that be better?

I'm not sure what would be the best option here (since we never validate for example that data len can't be u32::MAX I wouldn't go with an unwrap here :-?).

Reply:
done